arch/cortex-m/src/systick.rs

```diff
@@ -81,12 +85,28 @@ impl SysTick {
         res
     }
 
+    /// Initialize the `SysTick` with an explicit clock speed and external source
+    ///
+    /// Use this constructor if the core implementation does not have a
+    /// pre-calibration value and you need an external clock source for  
+    /// the Systick.
+    ///
+    ///   * `clock_speed` - the frequency of SysTick tics in Hertz. For example,
+    ///   if the SysTick is driven by the CPU clock, it is simply the CPU speed.
+    pub unsafe fn new_with_calibration_and_external_clock(clock_speed: u32) -> SysTick {
+        let mut res = SysTick::new();
+        res.hertz = clock_speed;
+        res.external_clock = true;
+        res
+    }
+
     // Return the tic frequency in hertz. If the calibration value is set in
     // hardware, use `self.hertz`, which is set in the `new_with_calibration`
-    // constructor.
+    // constructor. However, if there is value configured by the user, choose
+    //`self.hertz` instead.
     fn hertz(&self) -> u32 {
         let tenms = SYSTICK_BASE.syst_calib.read(CalibrationValue::TENMS);
-        if tenms == 0 {
+        if tenms == 0 || self.hertz != 0 {
```

Comment:
Something about the way the is commented / implemented is confusing. Given the comments, could this be simplified to just:

```rust
fn hertz(&self) -> u32 {
    if self.hertz != 0 {
         self.hertz
    } else {
         let tenms = SYSTICK_BASE.syst_calib.read(CalibrationValue::TENMS);
         tenms * 100
    }
}
```

?

Reply:
Yes and it is far more clear this way. I also updated the comment to reflect the implementation.